From e66f807701cd168e3a8ec545f747edca77d1950c Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Mon, 4 Mar 2024 13:34:23 +0100
Subject: [PATCH 4/4] conf: add card config for stm32mp25 dk board

Add ALSA card configuration file for STM32MP25 DK boards.

Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
---
 src/conf/cards/Makefile.am      |  3 ++-
 src/conf/cards/STM32MP25DK.conf | 16 ++++++++++++++++
 src/conf/cards/aliases.conf     |  1 +
 3 files changed, 19 insertions(+), 1 deletion(-)
 create mode 100644 src/conf/cards/STM32MP25DK.conf

diff --git a/src/conf/cards/Makefile.am b/src/conf/cards/Makefile.am
index 6812cdfd033b..4fd7308260f6 100644
--- a/src/conf/cards/Makefile.am
+++ b/src/conf/cards/Makefile.am
@@ -62,7 +62,8 @@ cfg_files = aliases.conf \
 	STM32MP15EV.conf \
 	STM32MP15DK.conf \
 	STM32MP13EV1.conf \
-	STM32MP25EV1.conf
+	STM32MP25EV1.conf \
+	STM32MP25DK.conf
 
 if BUILD_ALISP
 cfg_files += aliases.alisp
diff --git a/src/conf/cards/STM32MP25DK.conf b/src/conf/cards/STM32MP25DK.conf
new file mode 100644
index 000000000000..b6367c2b0f69
--- /dev/null
+++ b/src/conf/cards/STM32MP25DK.conf
@@ -0,0 +1,16 @@
+<confdir:pcm/hdmi.conf>
+
+STM32MP25DK.pcm.hdmi.0 {
+        @args [ CARD AES0 AES1 AES2 AES3 ]
+        @args.CARD { type string }
+        @args.AES0 { type integer }
+        @args.AES1 { type integer }
+        @args.AES2 { type integer }
+        @args.AES3 { type integer }
+        type asym
+        playback.pcm {
+                type hw
+                card $CARD
+                device 0
+        }
+}
diff --git a/src/conf/cards/aliases.conf b/src/conf/cards/aliases.conf
index 14f4f59059d2..2e2e8e2049dd 100644
--- a/src/conf/cards/aliases.conf
+++ b/src/conf/cards/aliases.conf
@@ -61,6 +61,7 @@ STM32MP15-EV cards.STM32MP15EV
 STM32MP15-DK cards.STM32MP15DK
 STM32MP13-EV1 cards.STM32MP13EV1
 STM32MP25-EV1 cards.STM32MP25EV1
+STM32MP25-DK cards.STM32MP25DK
 
 <confdir:ctl/default.conf>
 <confdir:pcm/default.conf>
-- 
2.25.1

